lib/fog/compute/google/requests/insert_server.rb

```diff
@@ -129,12 +129,22 @@ def handle_networks(options)
           end
 
           # Objectify the network if needed
-          unless network.is_a? Network
+          unless network.is_a?(Network)
             network = networks.get(network)
+            network_interfaces = network.get_as_interface_config(access_config)
+          end
+
+          if options.key? "subnetwork"
+            subnetwork = options.delete "subnetwork"
+            # Objectify the subnetwork if needed
+            unless subnetwork.is_a?(Subnetwork)
+              subnetwork = subnetworks.get(subnetwork, region_name)
+              network_interfaces = subnetwork.update_interface_config(network_interfaces)
+            end
           end
 
           # Return a networkInterfaces array
-          [network.get_as_interface_config(access_config)]
+          [network_interfaces]
```

Comment:
@Wyosotis does this help at all? https://github.com/fog/fog-google/blob/master/.fog.example